backend/src/test/java/gov/cdc/usds/simplereport/service/AzureStorageQueueTestEventReportingServiceTest.java

```diff
@@ -45,6 +46,21 @@ void dispatches_supplied_event_to_queue_client() {
     verify(client, times(1)).sendMessage(argThat(matcherForTest(testEvent)));
   }
 
+  @Test
+  void throws_custom_test_event_serialization_failure_exception() {
+    var client = mock(QueueAsyncClient.class);
+    Mono<SendMessageResult> response = mock(Mono.class);
+    when(response.toFuture())
+        .thenReturn(CompletableFuture.completedFuture(new SendMessageResult()));
+    when(client.sendMessage(any(String.class))).thenReturn(response);
+
+    var sut = new AzureStorageQueueTestEventReportingService(new ObjectMapper(), client);
+    var invalidTestEventWithNoResults = new TestEvent();
+    assertThrows(
+        TestEventSerializationFailureException.class,
+        () -> sut.report(invalidTestEventWithNoResults));
```

Comment:
To add coverage for the `TestEventSerializationFailureException` I think you should be able to get the error from the assertThrows like this 
```
var error = assertThrows(
        TestEventSerializationFailureException.class,
        () -> sut.report(invalidTestEventWithNoResults));
```
and then from there you can assert the other two methods on that class.

Reply:
+1, this will allow you to assert on the message thrown, etc. This class has some examples: 
https://github.com/CDCgov/prime-simplereport/blob/7abf48e7aeb582e62cc82f6b25591ccce3e44e67/backend/src/test/java/gov/cdc/usds/simplereport/idp/repository/LiveOktaRepositoryTest.java#L468-L468

Reply:
Done! Thank you both!!